packages/payment/src/services/payment-module.ts

```diff
@@ -160,305 +176,405 @@ export default class PaymentModuleService<
-      payment_collection: inputData.payment_collection_id,
-      payment_session: inputData.payment_session_id,
-      ...inputData,
-    }))
+    // TODO: what checks should be done here? e.g. captured_amount === amount?
 
-    const payments = await this.paymentService_.create(input, sharedContext)
+    const updated = await this.paymentCollectionService_.update(
+      input,
+      sharedContext
+    )
 
-    return await this.baseRepository_.serialize<PaymentDTO[]>(
-      Array.isArray(data) ? payments : payments[0],
-      {
-        populate: true,
-      }
+    return await this.baseRepository_.serialize(
+      Array.isArray(paymentCollectionId) ? updated : updated[0],
+      { populate: true }
     )
   }
 
-  updatePayment(
-    data: UpdatePaymentDTO,
-    sharedContext?: Context | undefined
-  ): Promise<PaymentDTO>
-  updatePayment(
-    data: UpdatePaymentDTO[],
-    sharedContext?: Context | undefined
-  ): Promise<PaymentDTO[]>
+  /**
+   *
+   * ********** PAYMENT SESSION **********
+   *
+   */
 
   @InjectTransactionManager("baseRepository_")
-  async updatePayment(
-    data: UpdatePaymentDTO | UpdatePaymentDTO[],
+  async createPaymentSession(
+    paymentCollectionId: string,
+    data: CreatePaymentSessionDTO,
     @MedusaContext() sharedContext?: Context
-  ): Promise<PaymentDTO | PaymentDTO[]> {
-    const input = Array.isArray(data) ? data : [data]
-    const result = await this.paymentService_.update(input, sharedContext)
+  ): Promise<PaymentSessionDTO> {
+    const sessionData = await this.paymentProviderService_.createSession(
+      data.provider_id,
+      data.providerContext
+    )
 
-    return await this.baseRepository_.serialize<PaymentDTO[]>(
-      Array.isArray(data) ? result : result[0],
+    const created = await this.paymentSessionService_.create(
       {
-        populate: true,
-      }
+        provider_id: data.provider_id,
+        amount: data.providerContext.amount,
+        currency_code: data.providerContext.currency_code,
+        payment_collection: paymentCollectionId,
+        data: sessionData,
+      },
+      sharedContext
     )
-  }
 
-  capturePayment(
-    data: CreateCaptureDTO,
-    sharedContext?: Context
-  ): Promise<PaymentDTO>
-  capturePayment(
-    data: CreateCaptureDTO[],
-    sharedContext?: Context
-  ): Promise<PaymentDTO[]>
+    return this.baseRepository_.serialize(created, { populate: true })
+  }
 
-  @InjectManager("baseRepository_")
-  async capturePayment(
-    data: CreateCaptureDTO | CreateCaptureDTO[],
-    @MedusaContext() sharedContext: Context = {}
-  ): Promise<PaymentDTO | PaymentDTO[]> {
-    const input = Array.isArray(data) ? data : [data]
+  @InjectTransactionManager("baseRepository_")
+  async updatePaymentSession(
+    data: UpdatePaymentSessionDTO,
+    @MedusaContext() sharedContext?: Context
+  ): Promise<PaymentSessionDTO> {
+    const session = await this.paymentSessionService_.retrieve(
+      data.id,
+      {},
```

Comment:
todo: probably a good idea to add a select to not load everything from the entity. It is a good practice in general